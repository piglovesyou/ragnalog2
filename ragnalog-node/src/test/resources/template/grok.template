in:
  type: file
  path_prefix: <input_file>
  parser:
    charset: UTF-8
    newline: CRLF
    type: grok
    grok_pattern: "%{APACHELOG}"
    grok_pattern_files:
<grok/grok_pattern_files:{file|
      - <file>
}>    timestamp_parser: sdf
    columns:
      - { name: bytes_sent , type: long }
      - { name: connection_status , type: string }
      - { name: http_version , type: string }
      - { name: logical_username , type: string }
      - { name: raw_request , type: string }
      - { name: referrer , type: string }
      - { name: remote_addr , type: string }
      - { name: request , type: string }
      - { name: response_time , type: long }
      - { name: status , type: long }
      - { name: timestamp , type: timestamp , format: "%d/%b/%Y:%T %z" }
      - { name: user_agent , type: string }
      - { name: username , type: string }
      - { name: verb , type: string }
filters:
  - type: ragnalog
    columns:
      extra: <extra>
out:
  type: elasticsearch
  nodes:
    - {host: <elasticsearch/host>, port: <elasticsearch/port> }
  cluster_name: <elasticsearch/cluster_name>
  bulk_actions: 30000
  index: <index_name>
  index_type: apache.access