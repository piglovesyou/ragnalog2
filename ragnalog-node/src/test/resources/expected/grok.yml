in:
  type: file
  path_prefix: /tmp/ragnalog/sample.log
  parser:
    charset: UTF-8
    newline: CRLF
    type: grok
    grok_pattern: "%{APACHELOG}"
    grok_pattern_files:
      - /home/ragnalog/grok-patterns
    timestamp_parser: sdf
    columns:
      - { name: bytes_sent , type: long }
      - { name: connection_status , type: string }
      - { name: http_version , type: string }
      - { name: logical_username , type: string }
      - { name: raw_request , type: string }
      - { name: referrer , type: string }
      - { name: remote_addr , type: string }
      - { name: request , type: string }
      - { name: response_time , type: long }
      - { name: status , type: long }
      - { name: timestamp , type: timestamp , format: "%d/%b/%Y:%T %z" }
      - { name: user_agent , type: string }
      - { name: username , type: string }
      - { name: verb , type: string }
filters:
  - type: insert
    columns:
      - extra: ap1
out:
  type: elasticsearch
  nodes:
    - {host: localhost, port: 9300 }
  cluster_name: ragnalog2.elasticsearch
  bulk_actions: 30000
  index: ragnalog-test
  index_type: apache.access